xapian-letor/api/ranker.cc

```diff
@@ -37,11 +37,15 @@
 using namespace std;
 using namespace Xapian;
 
-struct scoreComparer {
-    bool operator()(const pair<Xapian::docid, int>& first_pair, const pair<Xapian::docid, int>& second_pair) const {
-        return first_pair.second > second_pair.second;
-    }
-};
+bool
+Ranker::ScoreComparer::operator()(const FeatureVector & firstfv, const FeatureVector& secondfv) const {
+    return firstfv.get_score() > secondfv.get_score();
+}
```

Comment:
Is there a reason these would need to be overridden? I'd have thought just providing them as utility functions would be as helpful (and should take a little less memory).